camera.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2025 PHYTEC Messtechnik GmbH
+# SPDX-License-Identifier: Apache-2.0
+# Author: Martin Schwan <m.schwan@phytec.de>
+
+import cv2 as cv
+import glob
+import os
+import subprocess
+
+class Camera():
+    def __init__(self):
+        self.cameraDevice = None
+        self.colorConversionCode = None
+        self.apiPreference = None
+        self.videoCapture = cv.VideoCapture()
+
+    def open(self, filename):
+        if self.apiPreference is None:
+            raise AttributeError('API preference must be set before opening '
+                                 'video capture device!')
+        self.videoCapture.open(filename, self.apiPreference)
+        if not self.videoCapture.isOpened():
+            raise ValueError(f'Failed opening video capture device "{filename}"!')
+
+    def convert_frame_color(self, frame):
+        if self.colorConversionCode is None:
+            raise AttributeError('Color conversion code must be set before '
+                                 'converting frame colors!')
+        return cv.cvtColor(frame, self.colorConversionCode)
+
+class CameraUSB(Camera):
+    def __init__(self):
+        super().__init__()
+        self.colorConversionCode = cv.COLOR_BGR2RGB
+        self.apiPreference = cv.CAP_ANY
+
+    def open(self, filename=1):
+        super().open(filename)
+
+class CameraVM016(Camera):
+    def __init__(self):
+        super().__init__()
+        self.colorConversionCode = cv.COLOR_BAYER_GB2RGB
+        self.apiPreference = cv.CAP_GSTREAMER
+
+        if cv.getBuildInformation().find('GStreamer') < 0:
+            raise ValueError('This version of OpenCV does not support GStreamer!')
+
+    def open(self, filename=None):
+        if filename is None:
+            cameraDevices = glob.glob('/dev/cam-csi[0-9]*')
+            if not cameraDevices:
+                raise FileNotFoundError('No VM016 camera devices found!')
+            filename = cameraDevices[0]
```

Comment:
I don't think this would work reliably. You need to select `/dev/cam-csi1` here.
If someone would fiddle with the bootenv.txt and only CSI2 overlays are selected, `cameraDevices[0]` would contain `/dev/cam-csi2`.